packages/api/src/payments/paymentProvider.ts

```diff
@@ -240,6 +240,13 @@ export function setupPaymentProvider(opts: WepublishServerOpts): Router {
 
       const {items, ...invoiceData} = invoice
 
+      items.forEach(item => {
+        // Typing problem: Fix me @TODO
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
```

Comment:
@Itrulia do you have an idea how i could fix this?

Reply:
![image](https://user-images.githubusercontent.com/41843343/195645889-c9124f52-2236-4266-a52c-714072171ee9.png)

Reply:
```
const {items, ...invoiceData} = invoice

      await prisma.invoice.update({
        where: {id: invoice.id},
        data: {
          ...invoiceData,
          items: {
            deleteMany: {
              invoiceId: invoiceData.id
            },
            create: items.map(({invoiceId, ...item}) => item)
          },
          paidAt: new Date(),
          canceledAt: null
        }
      })
 ```
 
 should do the trick, also no mutation. Else we could also set invoiceId to null since the prisma model has it as nullable but not able to be undefined